qa/tasks/persistent_write_log_cache_thrash.py

```diff
@@ -0,0 +1,87 @@
+"""
+persistent write log cache thrash task
+"""
+import contextlib
+import logging
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.orchestra import run
+
+DEFAULT_KILL_DAEMON_TIME = 2
+DEFAULT_DEAD_TIME = 60
+DEFAULT_LIVE_TIME = 10
+
+log = logging.getLogger(__name__)
+
+@contextlib.contextmanager
+def thrashes_rbd_bench_on_persistent_cache(ctx, config):
+    """
+    thrashes rbd bench on persistent write log cache.
+    It can test recovery feature of persistent write log cache.
+    """
+    log.info("thrashes rbd bench on persistent write log cache")
+
+    client, client_config = list(config.items())[0]
+    (remote,) = ctx.cluster.only(client).remotes.keys()
+    client_config = client_config if client_config is not None else dict()
+    kill_daemon_time = client_config.get('kill_daemon_time', DEFAULT_KILL_DAEMON_TIME)
+    dead_time = client_config.get('dead_time', DEFAULT_DEAD_TIME)
+    live_time = client_config.get('live_time', DEFAULT_LIVE_TIME)
+    rbd_status_command = '"$(rbd status testimage | grep "Image cache state")"'
+
+    for i in range(kill_daemon_time):
+        log.info("start rbd-bench")
+        # rbd bench could not specify the run time so set a large enough test size.
+        remote.run(
+            args=[
+                'rbd', 'bench',
+                '--io-type', 'write',
+                '--io-pattern', 'rand',
+                '--io-total', '10240M',
+                'testimage'
+                ]
+            )
+        remote.run(
+                args=[
+                    'test', '-n', run.Raw(rbd_status_command),
+                    ]
+                )
+        remote.run(
+            args=[
+                'sleep', '{live_time}'.format(live_time=live_time),
+                ]
+            )
+        log.info("rbd bench crash....")
+        remote.run(
+            args=[
+                'sudo', 'killall', '-s', '9', 'rbd', run.Raw('||'), 'true',
+                ]
+            )
+        log.info("sleep....")
+        remote.run(
+            args=[
+                'sleep', '{dead_time}'.format(dead_time=dead_time),
+                ]
+            )
```

Comment:
Sleeping after `rbd bench` is killed is not necessary.  But we should assert that the cache is dirty, making sure that  `rbd bench` on the next iteration would go through the recovery path.